lib/src/signed_video_h26x_auth.c

```diff
@@ -668,27 +634,23 @@ verify_hashes_without_sei(signed_video_t *self)
       item = item->next;
       continue;
     }
-    // A new GOP starts if the NALU |is_first_nalu_in_gop|. Such a NALU is hashed twice; as an
-    // initial hash AND as a linking hash between GOPs. If this is the first time is is used in
-    // verification it also marks the start of a new GOP.
-    found_next_gop = item->nalu->is_first_nalu_in_gop && !item->need_second_verification;
-
-    // Mark the item as 'Not Authentic' or keep it for a second verification.
-    if (found_next_gop) {
-      // Keep the item pending and mark the first verification as not authentic.
-      item->need_second_verification = true;
-      item->first_verification_not_authentic = true;
-    } else if (item->validation_status == 'P') {
-      item->need_second_verification = false;
-      item->validation_status = 'N';
-      num_marked_items++;
+    // Stop verifying when SEI NALU is encountered. There can't be another NALU after SEI in the
+    // GOP.
+    if (item->nalu->is_gop_sei) {
+      break;
     }
+    item->validation_status = 'N';
+    num_marked_items++;
+
     item = item->next;
+    if (item) {
+      found_next_gop = item->nalu->is_first_nalu_in_gop && !item->used_for_linked_hash;
```

Comment:
Can you give an example when `found_next_gop == true` and at the same time is NOT pending.

Reply:
I don’t think there’s a situation where this could happen. We’re skipping all the NALUs that aren’t pending in line 633, and after setting the NALU status from P to N, we immediately check the next pending item to find the next GOP.

So I’m not really sure I understand what you’re asking or why you're concerned. Can you clarify what you mean?

Reply:
So this item will always be pending. Good.